apps/designer/app/designer/features/sidebar-left/panels/asset-manager.tsx

```diff
@@ -1,44 +1,49 @@
 import { ImageIcon } from "~/shared/icons";
-import { Button, Flex, Grid, Heading } from "~/shared/design-system";
-import { useRef } from "react";
-import { Form, useSubmit } from "@remix-run/react";
-import type { Asset } from "@webstudio-is/prisma-client";
+import { Flex, Grid, Heading } from "~/shared/design-system";
+import { useEffect, useState } from "react";
+import { useActionData } from "@remix-run/react";
+import { Asset } from "@webstudio-is/prisma-client";
 import { AssetManagerImage } from "./components/image";
 
-export const TabContent = ({ assets }: { assets: Array<Asset> }) => {
-  const inputRef = useRef<HTMLInputElement | null>(null);
-  const submit = useSubmit();
+import { AddAnAssetForm } from "./components/add-an-asset-form";
+import { UploadingAsset } from "../types";
+
+export const TabContent = ({
+  assets: baseAssets,
+}: {
+  assets: Array<Asset | UploadingAsset>;
+}) => {
+  const newImages = useActionData();
+
+  const [assets, setAsssets] = useState(baseAssets);
+
+  useEffect(() => {
+    if (newImages?.assets.length) {
+      setAsssets((currentAssets) => [
+        ...newImages.assets,
+        ...currentAssets.filter((asset) => !("uploading" in asset)),
+      ]);
+    }
+  }, [newImages]);
 
   return (
     <Flex gap="3" direction="column" css={{ padding: "$1" }}>
       <Flex justify="between">
         <Heading>Assets</Heading>
-        <Form
-          method="post"
-          encType="multipart/form-data"
-          onChange={(event) => {
-            if (inputRef.current?.files) {
-              submit(event.currentTarget);
-              event.currentTarget.reset();
-            }
-          }}
-        >
-          <input
-            accept="image/*"
-            type="file"
-            name="image"
-            multiple
-            ref={inputRef}
-            style={{ display: "none" }}
-          />
-          <Button type="button" onClick={() => inputRef?.current?.click()}>
-            Upload Image
-          </Button>
-        </Form>
+        <AddAnAssetForm
+          onSubmit={(uploadedAssets: Array<UploadingAsset>) =>
+            setAsssets((assets) => [...uploadedAssets, ...assets])
+          }
+        />
       </Flex>
       <Grid columns={2} gap={2}>
         {assets.map((asset) => (
-          <AssetManagerImage key={asset.id} asset={asset} />
+          <AssetManagerImage
+            key={asset.id}
+            path={asset.path}
+            alt={asset.alt || asset.name}
+            uploading={"uploading" in asset && asset.uploading}
```

Comment:
shouldn't it be asset.state or something with uploading one of the states and to be always defined?

Reply:
or asset.status

Reply:
Fixed this and used ststus